rusk-profile/src/lib.rs

```diff
@@ -207,45 +105,109 @@ pub fn verify_common_reference_string(buff: &[u8]) -> bool {
     hash == CRS_17
 }
 
-pub fn get_lockfile_path() -> PathBuf {
-    let mut dir = env::current_dir().expect("Failed to get current dir");
-    let mut lockfile = dir.join("Cargo.lock");
+pub fn clean_outdated_keys(ids: &Vec<[u8; 32]>) -> Result<(), io::Error> {
+    info!("Cleaning outdated keys (if any)");
+    let ids_as_str: Vec<String> =
+        ids.iter().map(|id| hex::encode(id)).collect();
+
+    let entries_to_delete = fs::read_dir(&get_rusk_keys_dir()?)?
+        .map(|res| res.map(|e| e.path()))
+        .filter(|res| res.is_ok())
+        .map(|res| res.unwrap())
+        .filter(|e| e.is_file())
+        .filter(|p| match p.extension() {
```

Comment:
Not a blocking nit, but this chain of matching patterns can be simplified into filter/map combinations